deep_qa/data/token_indexers/token_indexer.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict, List, Union
+
+from ...common.params import Params
+from ..vocabulary import Vocabulary
+
+
+TokenType = Union[int, List[int]]  # pylint: disable=invalid-name
```

Comment:
I think it makes the code below more readable.  Instead of wondering why the return type or input type of a method below is `Union[int, List[int]]`, you just know that the method takes or returns a token.  It's also like this because if you add a new token representation later, (e.g., List[List[int]], or Tuple[int, List[int]], or whatever, for whatever reason), and you want the types to remain accurate, you just modify this type definition.

Basically, the types are only there for readability - we don't actually use a type checker anywhere.  I think `TokenType` aids readability, and the complex `Union[]` type actually harms readability if it's there.  Does that make sense?